pkg/datapath/garp/events.go

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package garp
+
+import (
+	"context"
+	"net/netip"
+
+	"github.com/cilium/workerpool"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+var nodePodsCell = cell.Provide(func(lc hive.Lifecycle, c k8sClient.Clientset) (resource.Resource[*corev1.Pod], error) {
+	if !c.IsEnabled() {
+		return nil, nil
+	}
+
+	lw := utils.ListerWatcherWithFields(utils.ListerWatcherFromTyped[*corev1.PodList](
+		c.CoreV1().Pods("")), fields.OneTermEqualSelector("spec.nodeName", nodeTypes.GetName()),
+	)
+	return resource.New[*corev1.Pod](lc, lw), nil
+})
+
+// EventProcessorCell processes k8s pod events for the current node and
+// determines if a Gratuitous ARP packet needs to be sent.
+var EventProcessorCell = cell.Module(
+	"garp-event-processor",
+	"GARP event processor",
+
+	Cell,
+	nodePodsCell,
+
+	cell.Provide(newEventProcessor),
```

Comment:
Since the `EventProcessor` doesn't really provide an API for other modules we can use `cell.Invoke` instead. This will also make sure that it's strictly constructed.

Something along the lines of:

```
// cell.go:
var Cell = cell.Module(
  "garp",
  "Sends Gratuitous ARPs for local pods",

   cell.Config(Config{}),
   cell.Provide(newGARPSender),
   cell.Invoke(registerEventProcessor),
),

// events.go:
func registerEventProcessor(p eventProcessorParams) {
  ep := &eventProcessor{...}

  group := p.JobRegistry.NewGroup(...)
  group.Add(job.OneShot("garp-event-processor", ep.run))

  p.Lifecycle.Append(group)
}
```